Code/Engine/Texture/TexConv/Implementation/TextureAtlas.cpp

```diff
@@ -103,18 +116,21 @@ ezResult ezTexConvProcessor::LoadAtlasInputs(const ezTextureAtlasCreationDesc& a
 
       EZ_SUCCEED_OR_RETURN(ConvertAndScaleImage(srcItem.m_sAlphaInput, alphaImg, uiResX, uiResY));
 
+
+      // layer 0 must have the exact same size as the alpha texture
+      EZ_SUCCEED_OR_RETURN(ConvertAndScaleImage(srcItem.m_sLayerInput[0], item.m_InputImage[0], uiResX, uiResY));
+
+      // copy alpha channel into layer 0
+      ezImageUtils::CopyChannel(item.m_InputImage[0], 3, alphaImg, 0);
+
       // rescale all layers to be no larger than the alpha mask texture
-      for (ezUInt32 layer = 0; layer < atlasDesc.m_Layers.GetCount(); ++layer)
+      for (ezUInt32 layer = 1; layer < atlasDesc.m_Layers.GetCount(); ++layer)
```

Comment:
Is this rescaling really needed? Each layer has it's own rect in the atlas so they don't need to be the same size.

Reply:
Well, this only happens when you combine a separate alpha channel with the others. If you already have your alpha channel in the base color texture, this code path is not taken.
The (current) rationally for me is, that I mostly implemented the separate alpha channel to combine it with arbitrary other textures (e.g. ground textures). Those are typically very high resolution and would fill up the atlas quickly. Open to suggestions, though. Could also add an option per decal asset.